vendors/nuvoton/boards/numaker_iot_m487_wifi/ports/wifi/iot_wifi.c

```diff
@@ -376,21 +421,34 @@ WIFIReturnCode_t WIFI_GetPMMode( WIFIPMMode_t * pxPMModeType, void * pvOptionVal
 }
 /*-----------------------------------------------------------*/
 
-BaseType_t WIFI_IsConnected( void )
+BaseType_t WIFI_IsConnected( const WIFINetworkParams_t * pxNetworkParams )
 {
     BaseType_t xIsConnected = pdFALSE;
 
     if (xSemaphoreTake(xNuWiFi.xWifiSem, xSemaphoreWaitTicks) == pdTRUE) {
         if (ESP_WIFI_IsConnected(&xNuWiFi.xWifiObject) == pdTRUE) {
-            xIsConnected = pdTRUE;
+						if ( pxNetworkParams )
+						{
+								if( pxNetworkParams->ucSSIDLength > 0 
+										&& pxNetworkParams->ucSSIDLength <= wificonfigMAX_SSID_LEN
+										&& 0 == memcmp( pxNetworkParams->ucSSID, pcConnectedSSID, pxNetworkParams->ucSSIDLength )
+										&& pxNetworkParams->ucSSIDLength == strlen( pcConnectedSSID ) ) 
```

Comment:
Can the checks be replaced with a `strncmp( pxNetworkParams->ucSSID, pcConnectedSSID, wificonfigMAX_SSID_LEN )`